### PR TITLE
Fix change detection in messages

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

Fixing change detection in messages tab.

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
Hi @dianabarsan I _think_  I fixed this error. I've tested and don't see it happening again. But could you please have a look as well? Thank you

File: webapp/src/ts/modules/messages/messages.component.ts [hunk 1]
```diff
@@ -83,12 +75,14 @@ export class MessagesComponent implements OnInit, OnDestroy {
       loadingContent,
       error,
     ]) => {
+      this.selectedConversationId = selectedConversation?.id;
```
Comment (dianabarsan) [inline webapp/src/ts/modules/messages/messages.component.ts:78]: I don't think `selectedConversationId` needs to be an instance variable. Is doesn't appear to be used anywhere.

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Cool. Just a couple of small cleanup requests.

File: webapp/src/ts/modules/messages/messages.component.ts (hunk 1 above)
Comment (latin-panda) [reply webapp/src/ts/modules/messages/messages.component.ts]: Oh yes, that can be removed. Done!

File: webapp/src/ts/modules/messages/messages.component.ts [hunk 2]
```diff
@@ -45,30 +45,22 @@ export class MessagesComponent implements OnInit, OnDestroy {
...
   ngOnInit(): void {
-    this.subscribeToRouter();
     this.subscribeToStore();
     this.tourService.startIfNeeded(this.route.snapshot);
     this.updateConversations().then(() => this.displayFirstConversation(this.conversations));
     this.watchForChanges();
   }
 
   ngOnDestroy(): void {
+    this.reset();
```
[the remark this answers is not in the document]
Comment (latin-panda) [reply webapp/src/ts/modules/messages/messages.component.ts]: ah right, I removed it and tested again, still looks good. Done.

**Comment by latin-panda** [comment]
@dianabarsan  thanks for the review! I've applied the feedback, can you please have a look? 👍

**Review by dianabarsan — APPROVED** [review]
Nice.

**Pull request merged**